lib/internal/crypto/ml_kem.js

```diff
@@ -90,35 +91,28 @@ function mlKemExportKey(key, format) {
     switch (format) {
       case kWebCryptoKeyFormatRaw: {
         if (key[kKeyType] === 'private') {
-          return key[kKeyObject][kHandle].rawSeed().buffer;
+          return TypedArrayPrototypeGetBuffer(key[kKeyObject][kHandle].rawSeed());
         }
 
-        return key[kKeyObject][kHandle].rawPublicKey().buffer;
+        return TypedArrayPrototypeGetBuffer(key[kKeyObject][kHandle].rawPublicKey());
       }
       case kWebCryptoKeyFormatSPKI: {
-        return key[kKeyObject][kHandle].export(kKeyFormatDER, kWebCryptoKeyFormatSPKI).buffer;
+        return TypedArrayPrototypeGetBuffer(key[kKeyObject][kHandle].export(kKeyFormatDER, kWebCryptoKeyFormatSPKI));
       }
       case kWebCryptoKeyFormatPKCS8: {
         const seed = key[kKeyObject][kHandle].rawSeed();
-        const buffer = new Uint8Array(86);
-        buffer.set([
-          0x30, 0x54, 0x02, 0x01, 0x00, 0x30, 0x0B, 0x06,
+        const orc = {
+          '__proto__': null,
+          'ML-KEM-512': 0x01,
+          'ML-KEM-768': 0x02,
+          'ML-KEM-1024': 0x03,
+        }[key[kAlgorithm].name];
+        const buffer = new Uint8Array([
+          0x30, 0x54, 0x02, 0x01, 0x00, 0x30, 0x0b, 0x06,
           0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04,
-          0x04, 0x00, 0x04, 0x42, 0x80, 0x40,
-        ], 0);
-        switch (key[kAlgorithm].name) {
-          case 'ML-KEM-512':
-            buffer.set([0x01], 17);
-            break;
-          case 'ML-KEM-768':
-            buffer.set([0x02], 17);
-            break;
-          case 'ML-KEM-1024':
-            buffer.set([0x03], 17);
-            break;
-        }
-        buffer.set(seed, 22);
-        return buffer.buffer;
+          0x04, orc, 0x04, 0x42, 0x80, 0x40, ...seed,
```

Comment:
nit: using `TypedArrayPrototypeSet` is "safer" (i.e. can't be tampered with, while `Buffer.prototype[Symbol.iterator]` could be tampered with) and maybe more efficient

Reply:
umph, ok then

Reply:
reverted to TypedArrayPrototypeSet